src/Server.php

```diff
@@ -83,16 +130,22 @@ public function handleConnection($socket)
 
     public function getPort()
     {
+        if (!is_resource($this->master)) {
+            return null;
+        }
+
         $name = stream_socket_get_name($this->master, false);
 
         return (int) substr(strrchr($name, ':'), 1);
     }
 
     public function shutdown()
     {
-        $this->loop->removeStream($this->master);
-        fclose($this->master);
-        $this->removeAllListeners();
+        if (is_resource($this->master)) {
```

Comment:
Minor nitpicking (and personal preference): I'd like to see an early return here.

```php
if (!is_resource($this->master)) {
    return;
}

$this->loop->removeStream($this->master);
fclose($this->master);
$this->removeAllListeners();
```